libs/ml/src/core/node.cpp

```diff
@@ -114,10 +108,25 @@ OperationsCount Node<TensorType>::ChargeForward(
       throw std::runtime_error("Unable to lock weak pointer.");
     }
 
-    cost += input_node_ptr->ChargeForward(visited_nodes);
+    auto cost_and_outputshape = input_node_ptr->ChargeForward(visited_nodes);
```

Comment:
would be nice to have 1 or 2 extra comments to spell out the logic in this function